lib/translate/inquiry.cc

```diff
@@ -0,0 +1,308 @@
+      char formatted_hex_string[41];
+      for (int i = 4; i < result.page_length; i += 5) {
+        formatted_hex_string[i] = '_';
+      }
+      formatted_hex_string[result.page_length - 1] = '.';
+
+      // insert numbers
+      int pos = 0;
+      for (int i = 0; i < result.page_length - 1; i++) {
+        if (formatted_hex_string[i] != '_') {
+          formatted_hex_string[i] = hex_string[pos++];
+        }
+      }
+
+      memcpy(result.product_serial_number, formatted_hex_string,
+             result.page_length);
+    } else {
+      // 6.1.3.1.2
+      result.page_length = 20;
+
+      // convert 128-bit number into hex string
+      char hex_string[17];
+      sprintf(hex_string, "%08llx", identify_namespace_data.eui64);
+
+      // insert _ and . in the correct positions
+      char formatted_hex_string[21];
+      for (int i = 4; i < result.page_length; i += 5) {
+        formatted_hex_string[i] = '_';
+      }
+      formatted_hex_string[result.page_length - 1] = '.';
+
+      // insert numbers
+      int pos = 0;
+      for (int i = 0; i < result.page_length - 1; i++) {
+        if (formatted_hex_string[i] != '_') {
+          formatted_hex_string[i] = hex_string[pos++];
+        }
+      }
+
+      memcpy(result.product_serial_number, formatted_hex_string,
+             result.page_length);
+    }
+  } else {
+    if (nguid_nz) {
+      // 6.1.3.1.1
+      result.page_length = 40;
+
+      // convert 128-bit number into hex string
+      char hex_string[33];
+      sprintf(hex_string, "%08llx", identify_namespace_data.nguid[0]);
+      sprintf(&hex_string[16], "%08llx", identify_namespace_data.nguid[1]);
+
+      // insert _ and . in the correct positions
+      char formatted_hex_string[41];
+      for (int i = 4; i < result.page_length; i += 5) {
+        formatted_hex_string[i] = '_';
+      }
+      formatted_hex_string[result.page_length - 1] = '.';
+
+      // insert numbers
+      int pos = 0;
+      for (int i = 0; i < result.page_length - 1; i++) {
+        if (formatted_hex_string[i] != '_') {
+          formatted_hex_string[i] = hex_string[pos++];
+        }
+      }
+
+      memcpy(result.product_serial_number, formatted_hex_string,
+             result.page_length);
+    } else {
+      // 6.1.3.1.3
+      // valid for NVMe 1.0 devices only
+      // TODO?
+    }
+  }
+
+  return result;
+}
+
+scsi_defs::UnitSerialNumber BuildUnitSerialNumberVpd() {
+  nvme_defs::IdentifyNamespace identify_namespace_data =
+      NvmeIdentifyNamespace();
+  return TranslateUnitSerialNumberVpdResponse(identify_namespace_data);
+}
+
+// TODO: write return value to a buffer
+// Main logic engine for the Inquiry command
+void translate(absl::Span<const uint32_t> raw_cmd) {
+  std::optional<scsi_defs::InquiryCommand> opt_cmd = RawToScsiCommand(raw_cmd);
+  if (!opt_cmd.has_value()) return;
+
+  scsi_defs::InquiryCommand cmd = opt_cmd.value();
+  if (cmd.evpd) {
+    switch (cmd.page_code) {
```

Comment:
Adding a enum for page_code would make this piece more readable, since instead of using hex values `0x00`, etc, we could use the name of the code:

```
switch(MakePageCode(cmd.page_code)) {
 case kSupportedVpdPages:
  scsi_defs::SupportedVitalProductData result = BuildSupportedVpdPages();
  break;
 case kUnitSerialNumberVpd:
  ...
}
```